### ISSUE TITLE
[NOUVEL ARTICLE] Road-map et ressources necessaires à la création d'un bot Discord

### ISSUE THREAD

**Issue by CallMePixelMan** [NotANameServer/Not-a-Hub#40]
### Sujet de l'article

Bonjour tout le monde, j'espère que vous allez bien ! 😄

Après avoir discuté avec Martin de baptiste dans le salon #bot-discord, je suis venu à la conclusion que les ressources présentes dans le salon étaient relativement maigres et / ou ne permettaient pas de prendre une direction claire pour les nouveaux développeurs.
Pour cause, les messages épinglés traites essentiellement de détails dans certains wrapper, la commande `!devbots` est axée sur la théorie et la communication async avec l'API Discord (rien de spécifique au bot-dev et à ses défis), etc. 
En somme, aucune direction n'est prise tant le sujet de la communication avec l'API Discord est vaste et ne se limite pas au bot-dev.

Pour ces raisons, je trouve qu'il manque un outil qui permette aux personnes qui ont une idée, mais aucune connaissance de l'API et des technologies auxiliaires, de pouvoir se lancer en sachant quoi et où regarder (je rappelle que selon le règlement, tout ce qui est lié au bot-dev doit être fait dans le salon #bot-discord hors dérogation).

L'idée est la suivante; mettre à disposition un article permettant d'avoir accès, de manière centralisé; aux prérequis, aux technologies nécessaires en fonction des idées de la personne, aux documentations et aux guides liées à ces différentes technologies. 
Ainsi, l'accès à ces dernières est plus linéaire, la personne n'a plus besoin de demander de l'aide systématiquement pour des features qui font intervenir d'autres chose que simplement l'API et / ou le wrapper, et l'expérience développeur est en général plus agréable. 

***

Thread original : https://discord.com/channels/323076998576603137/1008712423257153598/1008715451028734092

**Comment by Iltotore** [comment]
Bonjour 👋 
> L'idée est la suivante; mettre à disposition un article permettant d'avoir accès, de manière centralisé; aux prérequis, aux technologies nécessaires en fonction des idées de la personne, aux documentations et aux guides liées à ces différentes technologies.

Dans cette phrase, tu parles toujours de ressources en lien avec la création de bots Discord ? Si oui je pense qu'il faudrait déjà ici dans un premier temps rassembler toutes les notions que tu voudrais intégrer dans l'article. Le sujet me parait trop vaste pour pouvoir être traité avec ses prérequis.

**Comment by CallMePixelMan** [comment]
> Dans cette phrase, tu parles toujours de ressources en lien avec la création de bots Discord ?

Oui, le but du salon reste d'aider les gens au sujet du dev de bots Discord.

Pour ce qui est des ressources, j'avais pensé à diviser l'article en plusieurs parties (il peut être même nécessaire de faire plusieurs articles en fonction de ce qui est abordé) : 
1 - TL;DR :ahde:
2 - Une présentation de la nomenclature
3 - A partir de la nomenclature, expliquer quel est le rôle du langage de prog, et du wrapper.
4 - Technologies auxiliaires et leur présentation (BDD, Serveurs Web, etc.).
5 - Bonne pratiques et connaissances avancées (logging, packaging, sharding, etc.)
6 - Annexe de guides divers sur les technologies (et référencés tout au long de l'article)

**Comment by mdevolde** [comment]
Coucou 🖐️

> Après avoir discuté avec Martin de baptiste dans le salon #bot-discord, je suis venu à la conclusion que les ressources présentes dans le salon étaient relativement maigres et / ou ne permettaient pas de prendre une direction claire pour les nouveaux développeurs.
> Pour cause, les messages épinglés traites essentiellement de détails dans certains wrapper, la commande !devbots est axée sur la théorie et la communication async avec l'API Discord (rien de spécifique au bot-dev et à ses défis), etc.
En somme, aucune direction n'est prise tant le sujet de la communication avec l'API Discord est vaste et ne se limite pas au bot-dev.

Si les messages épinglés traitent des différents wrappers, que la commande `!devbots` est axée théorie, c'est aussi parce qu'il n'existe selon moi pas de bon guide pour le développement de bot (ou alors, je ne l'ai pas encore trouvé). Donc, si tu désires faire un article qui ferait office de road map (et qui serait donc épinglé dans le salon #bots-discord), il faudrait évidemment que celui-ci soit de qualité, sur tous les sujets qu'il aborde.

> 4 - Technologies auxiliaires et leur présentation (BDD, Serveurs Web, etc.).
...
> 6 - Annexe de guides divers sur les technologies (et référencés tout au long de l'article)

Pour pouvoir prétendre présenter et donner des liens vers des guides reconnus à propos de ces différentes technos, il faut être sois même "expert" dans cette techno. Comment tu comptes avoir cette expertise dans ces différentes technos ? Je sais que tu ne comptes pas faire l'article seul, mais donc, il faudrait trouver des gens qui ont ces compétences recherchées (bien que pour certaines technos, comme pour les DB, tu pourrais rediriger vers le salon de NaN, où des ressources fiables sont déjà dispos).

Sinon, c'est plutôt une bonne initiative.

**Comment by CallMePixelMan** [comment]
A mon sens, s'il n'existe pas de bon tutoriel, c'est simplement parce que les gens n'apprennent pas à faire un bot, mais à coder une feature avec un wrapper (d'où la ribambelle de bots qui font tous la même chose, et seul quelques uns qui sortent un peu des sentiers battus pour proposer quelque chose de novateur).

L'objectif de l'article ce n'est pas de faire un tutoriel, mais de donner les outils en main aux nouveaux pour :
- Formuler leur idée ("J'ai envie de faire un bot Discord qui permette de garder des avis sur un service")
- Leur apprendre à déconstruire leur idée ("Pour ça il faut que je récupère des informations et que je les stockes..")
- Leur expliquer ce dont ils auront besoin pour utiliser Discord ("Ok, donc pour pouvoir envoyer des messages en tant que bot il me faut un wrapper, etc.")
- Leur expliquer quelle type de techno utiliser pour leurs défis ("Ok donc si je veux stocker des infos, il me faut une DB, par exemple sqlite")
- Leur link de la doc, des tutoriels pour les différents langages, etc. ("Ok donc je dois apprendre Python, d.py, et utiliser sqlite et je peux faire ça avec le tuto A et la doc B.")

Pour la deuxième partie de ton message, je pensais avoir quelque chose de communautaire, pas besoin d'être expert pour link un starting-guide vers MongoDB etc. l'objectif reste encore de juste mettre les gens "dans le droit chemin", pas de faire du tutorat.

**Comment by mdevolde** [comment]
Ok, je vois l'idée.

Par contre, ce que tu expliques, c'est tellement large. Je ne vois pas du tout comment tu comptes couvrir *tous/la plupart* des cas possibles. Tu dois soit faire un article super précis avec 1000 cas de figure, ou un article très vague, qui couvre le plus possible.
